java-spiffe-core/src/main/java/spiffe/workloadapi/retry/BackoffPolicy.java

```diff
@@ -0,0 +1,79 @@
+package spiffe.workloadapi.retry;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.val;
+
+import java.time.Duration;
+import java.util.function.UnaryOperator;
+
+@Data
+public class BackoffPolicy {
+
+    /**
+     * Retry indefinitely, default behavior
+     */
+    public static final int UNLIMITED_RETRIES = 0;
+
+    private static final int BACKOFF_MULTIPLIER = 2;
+
+    /**
+     * The first backoff delay period
+     */
+    Duration initialDelay = Duration.ofSeconds(1);
+
+    /**
+     * Max time of delay for the backoff period
+     */
+    Duration maxDelay = Duration.ofSeconds(60);
+
+    /**
+     * Max number of retries, unlimited by default
+     */
+    int maxRetries = UNLIMITED_RETRIES;
+
+    /**
+     * Function to calculate the backoff delay
+     */
+    UnaryOperator<Duration> backoffFunction = d -> d.multipliedBy(BACKOFF_MULTIPLIER);
+
+    /**
+     * Build backoff policy with defaults
+     */
+    public BackoffPolicy() {
+    }
+
+    @Builder
+    public BackoffPolicy(Duration initialDelay, Duration maxDelay, int maxRetries, UnaryOperator<Duration> backoffFunction) {
+        this.initialDelay = initialDelay != null ? initialDelay : Duration.ofSeconds(1);
+        this.maxDelay = maxDelay != null ? maxDelay : Duration.ofSeconds(60);
+        this.maxRetries = maxRetries;
+        this.backoffFunction = backoffFunction != null ? backoffFunction : d -> d.multipliedBy(BACKOFF_MULTIPLIER);
+    }
+
+    /**
+     * Calculate the nextDelay based on a currentDelay, applying the backoff function
+     * If the calculated delay is greater than maxDelay, it returns maxDelay
+     *
+     * @param currentDelay
+     * @return next delay
+     */
+    public Duration nextDelay(Duration currentDelay) {
+        val next = backoffFunction.apply(currentDelay);
+        if (next.compareTo(maxDelay) > 0) {
+            return maxDelay;
+        }
+        return next;
+    }
+
+    /**
+     * Returns true if the RetryPolicy is configure with UNLIMITED_RETRIES
+     * or if the retriesCount param is lower than the maxRetries
+     *
+     * @param retriesCount the current number of retries
+     * @return
+     */
+    public boolean doNotExceedMaxRetries(int retriesCount) {
```

Comment:
I found the name of this method a bit misleading. Let's suppose I have maxRetries = 3, then  doNotExceedMaxRetries(3) would return false when I would have expected true, because it doesn't exceed maxRetries.